mempool/src/core_mempool/transaction_store.rs

```diff
@@ -281,6 +282,9 @@ impl TransactionStore {
     pub(crate) fn get_required_balance(&mut self, address: &AccountAddress) -> u64 {
         self.transactions.get_mut(&address).map_or(0, |txns| {
             txns.iter().fold(0, |acc, (_, txn)| {
+                assume!(txn.gas_amount < u32::max_value() as u64); // seems more than reasonable
+                assume!(txn.txn.gas_unit_price() < u32::max_value() as u64);
```

Comment:
Similarly can this use `MAX_PRICE_PER_GAS_UNIT` instead?

Also could `txn.txn.gas_unit_price()` be bounded below by `MIN_PRICE_PER_GAS_UNIT`?